commit eb4a3c475c10ef0c7d65dce45fa179ce2c109e5d
Author: Grant <34662917+GQAssurance@users.noreply.github.com>
Date:   Fri Feb 15 14:46:26 2019 -0800

    Clear PoolManager in ‘remote_connection’ to ensure sockets are closed
    
    A fix for: https://github.com/SeleniumHQ/selenium/issues/6878
    
    When the ‘keep alive’ option is enabled the remote_connection object will create a PoolManager, but never deliberately clear it.  This can cause ResourceWarning warnings from open sockets.
    
    Added a new "close()" function which explicitly clears the pool.  This close function is called from Webdriver's “quit()” function (+1 squashed commit)

diff --git a/py/selenium/webdriver/remote/remote_connection.py b/py/selenium/webdriver/remote/remote_connection.py
index a9091eb8ae..9098bb5f23 100644
--- a/py/selenium/webdriver/remote/remote_connection.py
+++ b/py/selenium/webdriver/remote/remote_connection.py
@@ -440,3 +440,10 @@ class RemoteConnection(object):
         finally:
             LOGGER.debug("Finished Request")
             resp.close()
+
+    def close(self):
+        """
+        Clean up resources when finished with the remote_connection
+        """
+        if hasattr(self, '_conn'):
+            self._conn.clear()
diff --git a/py/selenium/webdriver/remote/webdriver.py b/py/selenium/webdriver/remote/webdriver.py
index 562adcb78d..565266f6b8 100644
--- a/py/selenium/webdriver/remote/webdriver.py
+++ b/py/selenium/webdriver/remote/webdriver.py
@@ -740,6 +740,7 @@ class WebDriver(object):
             self.execute(Command.QUIT)
         finally:
             self.stop_client()
+            self.command_executor.close()
 
     @property
     def current_window_handle(self):

commit 404c800d1eceb31d2b552b6594f8c183e57c1d2f
Author: GQAssurance <34662917+GQAssurance@users.noreply.github.com>
Date:   Thu Aug 8 16:05:54 2019 -0700

    Explicitly clear PoolManager by using context manager
    
    Fixes #6878
    
    This is meant to solve ResourceWarning messages when WebDrivers are instanced with a KeepAlive=False.

diff --git a/py/selenium/webdriver/remote/remote_connection.py b/py/selenium/webdriver/remote/remote_connection.py
index 58804e462f..ac9e0e6b4a 100644
--- a/py/selenium/webdriver/remote/remote_connection.py
+++ b/py/selenium/webdriver/remote/remote_connection.py
@@ -371,8 +371,8 @@ class RemoteConnection(object):
 
             statuscode = resp.status
         else:
-            http = urllib3.PoolManager(timeout=self._timeout)
-            resp = http.request(method, url, body=body, headers=headers)
+            with urllib3.PoolManager(timeout=self._timeout) as http:
+                resp = http.request(method, url, body=body, headers=headers)
 
             statuscode = resp.status
             if not hasattr(resp, 'getheader'):
